cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/analysis/MultiValueEvaluator.kt

```diff
@@ -92,6 +94,43 @@ class MultiValueEvaluator : ValueEvaluator() {
         return cannotEvaluate(node, this)
     }
 
+    /**
+     * We are handling some basic arithmetic compound assignment operations and string operations
+     * that are more or less language-independent.
+     */
+    override fun handleAssignExpression(node: AssignExpression, depth: Int): Any? {
+        // This only works for compound assignments
+        if (!node.isCompoundAssignment) {
+            return super.handleAssignExpression(node, depth)
+        }
+
+        // Resolve lhs
+        val lhsValue = evaluateInternal(node.lhs.singleOrNull(), depth + 1)
+        // Resolve rhs
+        val rhsValue = evaluateInternal(node.rhs.singleOrNull(), depth + 1)
+
+        if (lhsValue !is Collection<*> && rhsValue !is Collection<*>) {
+            return computeBinaryOpEffect(lhsValue, rhsValue, node)
+        }
+
+        val result = mutableSetOf<Any?>()
+        if (lhsValue is Collection<*>) {
+            for (lhs in lhsValue) {
+                if (rhsValue is Collection<*>) {
+                    result.addAll(rhsValue.map { r -> computeBinaryOpEffect(lhs, r, node) })
+                } else {
+                    result.add(computeBinaryOpEffect(lhs, rhsValue, node))
+                }
+            }
+        } else {
+            result.addAll(
+                (rhsValue as Collection<*>).map { r -> computeBinaryOpEffect(lhsValue, r, node) }
```

Comment:
a comment would be nice here. this is a bit confusing at first.
// `rhsValue` is a `Collection<*>` because of the check above and because `lhsValue` is not a `Collection<*>`

Reply:
This is copy paste code from the value evaluators magic wizard @KuechA ;) Can you maybe add a comment here, @KuechA ?

Reply:
I added some comments to explain this craziness